packages/rpc-core/src/rpc-methods/__typetests__/get-block-type-test.ts

```diff
@@ -0,0 +1,1256 @@
+import { Base58EncodedAddress } from '@solana/addresses';
+import { Blockhash, TransactionVersion } from '@solana/transactions';
+import assert from 'assert';
+
+import { TransactionError } from '../../transaction-error';
+import {
+    Base58EncodedBytes,
+    Base58EncodedDataResponse,
+    Base64EncodedDataResponse,
+    LamportsUnsafeBeyond2Pow53Minus1,
+    Reward,
+    TokenBalance,
+    TransactionStatus,
+    U64UnsafeBeyond2Pow53Minus1,
+} from '../common';
+import { mockTypeTestResponse, mockTypeTestRpc } from './common';
+
+const rpc = mockTypeTestRpc();
+
+function assertBase(
+    response: {
+        blockhash: string;
+        blockHeight: bigint;
+        blockTime: number;
+        parentSlot: bigint;
+        previousBlockhash: string;
+    } & { [key: string]: unknown }
+) {
+    response.blockhash satisfies string;
+    response.blockHeight satisfies bigint;
+    response.blockTime satisfies number;
+    response.parentSlot satisfies bigint;
+    response.previousBlockhash satisfies string;
+}
+
+// First overload
+// Rewards set to `false`
+{
+    const response = mockTypeTestResponse(
+        rpc.getBlock(0n, {
+            // No extra configs
+            rewards: false,
+            transactionDetails: 'none',
+        })
+    );
+    assert(response);
```

Comment:
`assertBase()` sort of covers this case too, right?

What about just doing;
```suggestion
    response satisfies object;
```

Reply:
Unfortunately TS barks at the `| null` possibility:

```
Type 'null' is not assignable to type 'object'
```

You also can't apparently just add `assert(response)` to the `assertBase` function, it won't satisfy TS either for other checks :(

Reply:
Ohhhh. I see what you're trying to do now.

Alright, well consider these alternatives:

```ts
const response = await rpc.foo().send();
if (response) {
  // Your type assertions.
}
```

…or:

```ts
function nonNullable<T>(_: T): NonNullable<T> {}

const response = nonNullable(
  await rpc.foo().send(),
);
// Your type assertions.
```

I like the first one.